### PR TITLE
feat(argo-cd): Argo CD 1.8.1

### PR DESCRIPTION
**Pull request by Junnplus**
thx for https://github.com/argoproj/argo-helm/pull/502
- Update argo-cd to 1.8.1
- Update 4.5.3 redis-ha helm https://github.com/argoproj/argo-cd/pull/3370
- Update redis to 5.0.10 https://github.com/argoproj/argo-cd/pull/4767
- Upgrade Dex to v2.25.0 https://github.com/argoproj/argo-cd/pull/4470

Checklist:

* [x] I have update the chart version in `Chart.yaml` following Semantic Versioning.
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have followed the testing instructions in the [contributing guide](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md).
* [x] I have signed the CLA and the build is green.
* [x] I will test my changes again once merged to master and published.

Changes are automatically published when merged to `master`. They are not published on branches.

### CODE REVIEW

**Comment by james-d-elliott** [comment]
I would recommend updating Dex IDP all the way to v2.27.0 due to the latest CVE affecting it via go's encoding/xml package.

File: charts/argo-cd/templates/argocd-application-controller/statefulset.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {{- $redisHa := (index .Values "redis-ha") -}}
 apiVersion: apps/v1
-kind: Deployment
+kind: StatefulSet
```
Comment (seanson) [inline charts/argo-cd/templates/argocd-application-controller/statefulset.yaml:3]: We've changed the way the StatefulSet will roll out with a feature flag until chart 3.x, could you please rebase this against master and fold the conditional in.

**Review by seanson — CHANGES_REQUESTED** [review]
Thanks for the PR @Junnplus! This looks pretty good, just one issue of note regarding feature flagging for the Stateful set and it should be good to go.

File: charts/argo-cd/templates/argocd-application-controller/statefulset.yaml (hunk 1 above)
Comment (Junnplus) [reply charts/argo-cd/templates/argocd-application-controller/statefulset.yaml:3]: Thx you for review. 
I think we can upgrade chart 3.0 for argocd 1.8, use StatefulSet for default.
> The argocd-application-controller converted to StatefulSet
in [1.7 to 1.8](https://argoproj.github.io/argo-cd/operator-manual/upgrading/1.7-1.8/ )

**Comment by Junnplus** [comment]
@james-d-elliott thank you for recommend, I think we can make a pr for upgrade dex to v2.27 individual.
Of course, i also upgrade in this.

File: charts/argo-cd/templates/argocd-application-controller/statefulset.yaml (hunk 1 above)
Comment (seanson) [reply charts/argo-cd/templates/argocd-application-controller/statefulset.yaml:3]: Hi @Junnplus - I'd like this to be conditional for a while first. Please restore the conditional `kind` and `serviceName`.

Comment (Junnplus) [reply charts/argo-cd/templates/argocd-application-controller/statefulset.yaml]: @seanson done.

**Comment by spencergilbert** [comment]
@Junnplus can you also signoff to fix the last check? https://github.com/argoproj/argo-helm/pull/520/checks?check_run_id=1563691451

**Comment by Junnplus** [comment]
@spencergilbert I fixed signoff, but has new failed for publish ci.

**Comment by cabrinha** [comment]
> @spencergilbert I fixed signoff, but has new failed for publish ci.

I think publish is failing due to the `requirements.yaml` in the argo chart: https://github.com/argoproj/argo-helm/pull/495#issuecomment-747272587

**Comment by spencergilbert** [comment]
> > @spencergilbert I fixed signoff, but has new failed for publish ci.
> 
> I think publish is failing due to the `requirements.yaml` in the argo chart: [#495 (comment)](https://github.com/argoproj/argo-helm/pull/495#issuecomment-747272587)

Last time I was playing with Helm that was still a warning not an error - has that changed? I'll try and look at it later tonight or tomorrow.

**Comment by cabrinha** [comment]
> > > @spencergilbert I fixed signoff, but has new failed for publish ci.
> > 
> > 
> > I think publish is failing due to the `requirements.yaml` in the argo chart: [#495 (comment)](https://github.com/argoproj/argo-helm/pull/495#issuecomment-747272587)
> 
> Last time I was playing with Helm that was still a warning not an error - has that changed? I'll try and look at it later tonight or tomorrow.

Looks like this commit caused the CI workflow to pass: https://github.com/argoproj/argo-helm/commit/aa863224ae8cfb8205a3ab02036708ba860a7bc1

File: charts/argo-cd/values.yaml [hunk 2]
```diff
@@ -28,15 +28,15 @@ controller:
...
+    tag: # v1.8.1
     imagePullPolicy: # IfNotPresent
 
   # If changing the number of replicas you must pass the number as ARGOCD_CONTROLLER_REPLICAS as an environment variable
   replicas: 1
 
   # Deploy the application as a StatefulSet instead of a Deployment, this is required for HA capability.
   # This is a feature flag that will become the default in chart version 3.x
-  enableStatefulSet: false
+  enableStatefulSet: true
```
Comment (seanson) [inline charts/argo-cd/values.yaml:39]: Please ensure this default is set to false.

**Comment by michaelajr** [comment]
Any reason this closed? Was hoping to get a chart bump to v3 and app bump to 1.8.x this week.

**Comment by Junnplus** [comment]
@michaelajr 2.11.0 already support StatefulSet for argocd 1.8, you can use `enableStatefulSet` and `image.tag` for upgrade to argocd 1.8

**Pull request closed without merging**